### PR TITLE
st_things: update st_things sample app

### PR DESCRIPTION
**Pull request by an4967**
- Add 2 dimmer resources
- Add handlers for additional resources

### CODE REVIEW

File: apps/examples/st_things/sample_light_things.c [hunk 1]
```diff
@@ -27,10 +27,16 @@
...
 #define TAG		"LIGHT_THINGS"
 
-static const char *g_res_switch = "/switch";
+static const char *g_res_switch = "/switch/main/0";
+static const char *g_res_dimming = "/switchLevel/main/0";
+static const char *g_res_temp = "/colorTemperature/main/0";
 
 extern bool handle_get_request_on_switch(st_things_get_request_message_s * req_msg, st_things_representation_s * resp_rep);
 extern bool handle_set_request_on_switch(st_things_set_request_message_s * req_msg, st_things_representation_s * resp_rep);
+extern bool handle_get_request_on_dimming(st_things_get_request_message_s * req_msg, st_things_representation_s * resp_rep);
```
Comment (sunghan-chang) [inline apps/examples/st_things/sample_light_things.c]: remove spaces
```
*req_msg
*resp_req
```

File: apps/examples/st_things/sample_light_user.c [hunk 2]
```diff
@@ -46,27 +52,72 @@ bool handle_get_request_on_switch(st_things_get_request_message_s *req_msg, st_t
...
+	return true;
+}
 
-		resp_rep->set_bool_value(resp_rep, PROPERTY_VALUE, g_switch_value);
-	}
+bool handle_set_request_on_dimming(st_things_set_request_message_s * req_msg, st_things_representation_s * resp_rep)
+{
+	printf("[%s]IN-handle_set_request_on_dimming() called..\n", TAG);
+
+    if (req_msg->rep->get_int_value(req_msg->rep, PROPERTY_VALUE_DIMMING, &g_dimming_setting))
```
Comment (sunghan-chang) [inline apps/examples/st_things/sample_light_user.c]: check indentation

File: apps/examples/st_things/sample_light_user.c [hunk 3]
```diff
@@ -26,17 +26,23 @@
...
 
-static const char *PROPERTY_VALUE = "value";
+static const char *PROPERTY_VALUE_SWITCH = "power";
+static const char *PROPERTY_VALUE_DIMMING = "dimmingSetting";
+static const char *PROPERTY_VALUE_COLOR_TEMPERATURE = "ct";
 
-static bool g_switch_value = true;
+static char *power_status[] = {"on", "off"};
+
+bool g_switch_value = false;
```
Comment (sunghan-chang) [inline apps/examples/st_things/sample_light_user.c]: why do you remove "static"? These three global variables use only in this file.

**Comment by an4967** [comment]
@sunghan-chang I applied your requests.

File: apps/examples/st_things/sample_light_things.c [hunk 4]
```diff
@@ -70,31 +76,34 @@ static bool handle_ownership_transfer_request(void)
...
 {
 	printf("Received a SET request on %s\n", req_msg->resource_uri);
 
 	if (0 == strncmp(req_msg->resource_uri, g_res_switch, strlen(g_res_switch))) {
-		handle_set_request_on_switch(req_msg, resp_rep);
-		return true;
-	} else {
-		return false;
+		return handle_set_request_on_switch(req_msg, resp_rep);	
+	}else if (0 == strncmp(req_msg->resource_uri, g_res_dimming, strlen(g_res_dimming))) {
```
Comment (sunghan-chang) [inline apps/examples/st_things/sample_light_things.c]: add a space 
```
} else
```

File: apps/examples/st_things/sample_light_things.c [hunk 5]
```diff
@@ -70,31 +76,34 @@ static bool handle_ownership_transfer_request(void)
...
+		return handle_set_request_on_dimming(req_msg, resp_rep);
+	}else if (0 == strncmp(req_msg->resource_uri, g_res_temp, strlen(g_res_temp))) {
```
Comment (sunghan-chang) [inline apps/examples/st_things/sample_light_things.c]: same

File: apps/examples/st_things/sample_light_user.c [hunk 6]
```diff
@@ -47,26 +53,71 @@ bool handle_set_request_on_switch(st_things_set_request_message_s *req_msg, st_t
 {
 	printf("[%s]IN-handle_set_request_on_switch() called..\n", TAG);
 
-	bool is_value_changed = false;
+	char *power;
+	if (req_msg->rep->get_str_value(req_msg->rep, PROPERTY_VALUE_SWITCH, &power)){
```
Comment (sunghan-chang) [inline apps/examples/st_things/sample_light_user.c]: add a space
```
) {
```

File: apps/examples/st_things/sample_light_user.c [hunk 7]
```diff
@@ -47,26 +53,71 @@ bool handle_set_request_on_switch(st_things_set_request_message_s *req_msg, st_t
 {
 	printf("[%s]IN-handle_set_request_on_switch() called..\n", TAG);
 
-	bool is_value_changed = false;
+	char *power;
+	if (req_msg->rep->get_str_value(req_msg->rep, PROPERTY_VALUE_SWITCH, &power)){	
+		printf("[%s] power : %s\n", TAG, power);
+		if(strcmp(power, "off") == 0){
```
Comment (sunghan-chang) [inline apps/examples/st_things/sample_light_user.c]: space

File: apps/examples/st_things/sample_light_user.c [hunk 8]
```diff
@@ -47,26 +53,71 @@ bool handle_set_request_on_switch(st_things_set_request_message_s *req_msg, st_t
...
+			g_switch_value = false;
+		}else{
```
Comment (sunghan-chang) [inline apps/examples/st_things/sample_light_user.c]: same

File: apps/examples/st_things/sample_light_user.c [hunk 9]
```diff
@@ -47,26 +53,71 @@ bool handle_set_request_on_switch(st_things_set_request_message_s *req_msg, st_t
...
+	printf("[%s]OUT-handle_get_request_on_dimming() called..\n", TAG);
+	return true;
+}
 
-		resp_rep->set_bool_value(resp_rep, PROPERTY_VALUE, g_switch_value);
+bool handle_set_request_on_dimming(st_things_set_request_message_s * req_msg, st_things_representation_s * resp_rep)
+{
+	printf("[%s]IN-handle_set_request_on_dimming() called..\n", TAG);
+
+	if (req_msg->rep->get_int_value(req_msg->rep, PROPERTY_VALUE_DIMMING, &g_dimming_setting))
```
Comment (sunghan-chang) [inline apps/examples/st_things/sample_light_user.c]: location of opening bracket

File: apps/examples/st_things/sample_light_user.c [hunk 10]
```diff
@@ -47,26 +53,71 @@ bool handle_set_request_on_switch(st_things_set_request_message_s *req_msg, st_t
...
+	printf("[%s]OUT-handle_set_request_on_switch() called..\n", TAG);
+	return true;
+}
 
-		if (g_switch_value != desired_value) {
-			g_switch_value = desired_value;
-			printf("[%s]changed switch value: %d\n", TAG, g_switch_value);
-			is_value_changed = true;
-		}
+bool handle_get_request_on_dimming(st_things_get_request_message_s * req_msg, st_things_representation_s * resp_rep)
```
Comment (sunghan-chang) [inline apps/examples/st_things/sample_light_user.c]: remove spaces

File: apps/examples/st_things/sample_light_user.c [hunk 11]
```diff
@@ -47,26 +53,71 @@ bool handle_set_request_on_switch(st_things_set_request_message_s *req_msg, st_t
...
+	if (req_msg->has_property_key(req_msg, PROPERTY_VALUE_DIMMING)) {
+		resp_rep->set_int_value(resp_rep, PROPERTY_VALUE_DIMMING, g_dimming_setting);
+	}
+
+	printf("[%s]OUT-handle_get_request_on_dimming() called..\n", TAG);
+	return true;
+}
 
-		resp_rep->set_bool_value(resp_rep, PROPERTY_VALUE, g_switch_value);
+bool handle_set_request_on_dimming(st_things_set_request_message_s * req_msg, st_things_representation_s * resp_rep)
```
Comment (sunghan-chang) [inline apps/examples/st_things/sample_light_user.c]: spaces

File: apps/examples/st_things/sample_light_user.c [hunk 12]
```diff
@@ -47,26 +53,71 @@ bool handle_set_request_on_switch(st_things_set_request_message_s *req_msg, st_t
...
+		printf("dimming : %lld\n", g_dimming_setting);
+		resp_rep->set_int_value(resp_rep, PROPERTY_VALUE_DIMMING, g_dimming_setting);
+		st_things_notify_observers(req_msg->resource_uri);
+	}	
+
+	printf("[%s]OUT-handle_set_request_on_dimming() called..\n", TAG);
+	return true;
+}
+
+bool handle_get_request_on_ct(st_things_get_request_message_s * req_msg, st_things_representation_s * resp_rep)
```
Comment (sunghan-chang) [inline apps/examples/st_things/sample_light_user.c]: spaces

File: apps/examples/st_things/sample_light_user.c [hunk 13]
```diff
@@ -47,26 +53,71 @@ bool handle_set_request_on_switch(st_things_set_request_message_s *req_msg, st_t
...
+	if (req_msg->has_property_key(req_msg, PROPERTY_VALUE_COLOR_TEMPERATURE)) {
+		resp_rep->set_int_value(resp_rep, PROPERTY_VALUE_COLOR_TEMPERATURE, g_color_temp);
 	}
 
-	if (is_value_changed) {
+	printf("[%s]OUT-handle_get_request_on_ct() called..\n", TAG);
+	return true;
+}
+
+bool handle_set_request_on_ct(st_things_set_request_message_s * req_msg, st_things_representation_s * resp_rep)
```
Comment (sunghan-chang) [inline apps/examples/st_things/sample_light_user.c]: spaces

**Comment by an4967** [comment]
@sunghan-chang I applied your requests.

**Pull request merged**